EventServer/eventserver.go

```diff
@@ -50,6 +51,48 @@ func NewEventServer() *EventServer {
 	}
 }
 
+var blacklistedUserAgents = []string{
+	"Python",                // Python scripts
+	"curl",                  // cURL
+	"Postman",               // Postman API client
+	"HttpClient",            // Generic HTTP client
+	"Java",                  // Java clients
+	"Go-http-client",        // Go's default HTTP client
+	"Wget",                  // Wget utility
+	"php",                   // PHP scripts
+	"Ruby",                  // Ruby scripts
+	"Node.js",               // Node.js scripts
+	"BinGet",                // BinGet utility
+	"libwww-perl",           // Perl library
+	"Microsoft URL Control", // Microsoft URL Control tool
+	"Peach",                 // Peach fuzzing tool
+	"pxyscand",              // Proxy scanner
+	"PycURL",                // Python binding to libcurl
+	"Python-urllib",         // Python urllib library
+}
+
+func UserAgentBlacklist() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		userAgent := c.GetHeader("User-Agent")
+		if isBlacklisted(userAgent) {
+			// Block the request
+			c.JSON(http.StatusForbidden, gin.H{"message": "Blocked: Disallowed User-Agent"})
```

Comment:
You can combine c.JSON() and c.Abort() by using c.AbortWithStatusJSON().
For a use case example, you can refer to line 134